enhancements/network/ovn-hypershift-live-migration.md

```diff
@@ -0,0 +1,319 @@
+- No TCP connection breakage over E/W and N/S during live migration with minimal disruption.
+  - Disruption for pre-copy live migration ~1sec
+  - Disruption for post-copy live migration (this happend if libvirt activate the target domain before live migration finish) ~8sec
+- Expose hosted cluster pods running at workers (VMs) using NodePort/LoadBalancer services at infra cluster.
+- Allow workers services access to services like kubedns and apiserver from hosted control plane.
+
+To accomplish the goals listed above the ovn-kubernetes default network has to implement some features triggered for an Hypershift KubeVirt VM.
+
+## Motivation
+
+Hypershift and kubevirt integration requires the ability to have live-migration functionality across the network with OVN-Kubernetes implemented at the pod's default network.
+
+### User Stories
+
+I, a Hypershift user, need to migrate one hosted cluster without affecting different ones and have minimal
+disruption.
+
+### Goals
+
+Hypershift kubevirt provider needs the live-migration functionality, missing features at CNI need to be implement at the ovn-kuberentes 
+which is the default CNI for openshift, this should work with dual stack.
+
+### Non-Goals
+
+This feature is not implemented at secondary networks, only primary networks will be live migratable.
+
+## Proposal
+
+At kubevirt every VM is executed inside a "virt-launcher" pod; how the "virt-launcher" pod net interface is "bound" to the VM is specified by the network binding at the VM spec.
+
+The main VM network binding mechanisms are:
+- masquerade binding: the VM has a stable fake address that is masqueraded (SNAT) at virt-launcher pod. 
+- bridge binding: The virt-launcher interface is replaced by a bridge with one leg on the pod interface and the other at the VM tap device. When IPAM is configured on the pod interface, a DHCP server is started to advertise the pod's IP to DHCP aware VMs.
+
+Since Masquerade binding has a fake address at VM guest it cannot be used for hypershift, the kubernetes nodes should see the address they are using to register to the api server.
+
+The bridge binding is able to expose the pod IP to the VM as expected by hypershift but kubevirt does not support live migration with it, for kubevirt to support live migration with bridge binding 
+at primary interface the OVN kubernetes CNI and controllers need to do the following:
+- Do IP assignemnt with ovn-k but skip the CNI part that configure it at pod's netns veth.
+- Do not expect the IP to be on the pod netns.
+- Add the ability at ovn-k controllers to migrate pod's IP from node to node.
+
+
+
+### Workflow Description
+
+The gist of the design is:
+
+1. Hypershift kubevirt workers attach to ovn-k node switches but CNI does not set IP addresses in virt-launcher pods netns
+2. Deliver IP address at VMs configuring logical switch port DHCP options, the kubevirt DHCP server is deactivated.
```

Comment:
Could you elaborate a bit here ? Why do we need to deactivate KubeVirt's DHCP server ?...

AFAIU, it basically advertises the IP / MAC / MTU / routes of the pod's default network interface.

Reply:
We don't explicit deactivate it, the fact that we don't configure an IP at virt launcher pod's veth means that the kubevirt DHCP server do not advertise anything.